src/structs.zig

```diff
@@ -51,4 +51,13 @@ pub const Event = union(enum) {
     key_release: struct {
         key: enums.Key,
     },
+    mouse_motion: struct {
+        // These are in window coordinates (not framebuffer coords)
+        x: f64,
+        y: f64,
+    },
+    mouse_button: struct {
+        button: enums.MouseButton,
+        action: enums.Action,
```

Comment:
I m wondering if using Action is the right thing to do versus separate events for press and release (button_press, button_release) like we have for key events. The key events has been named so because many libraries choose to differentiate the events completely instead of having one event type for the category.

Reply:
Yes I'm also unsure.  In my gui library I started with mouse events being separated (leftdown, leftup, rightdown, rightup, etc.) and am moving back towards the glfw model with separate Action.  It seems to align the event handlers with how I think of them ("I'm handling mouse events here, then split by press/release).

But I don't think it matters that much.  I'll switch it.  Thanks!

Reply:
Also it matters a lot more that we are consistent.